tests/cpgrid/lgr/LgrChecks.hpp

```diff
@@ -239,26 +257,40 @@ void Opm::checkGlobalCellBoundsConsistencyLevelZeroAndLeaf(const std::vector<int
 void Opm::checkFatherAndSiblings(const Dune::cpgrid::Entity<0>& element,
                                  int preAdaptMaxLevel,
                                  double expected_total_children,
-                                 const Dune::CpGrid& grid)
+                                 const Dune::CpGrid& grid,
+                                 bool isNested)
 {
-    const auto& father = element.father();
-    const auto& fatherLevelData = grid.currentData()[father.level()];
-
-    BOOST_CHECK( father.level() <= preAdaptMaxLevel );
-    BOOST_CHECK_EQUAL( fatherLevelData ->getMark(father), 1);
-    BOOST_CHECK( !father.isLeaf()); // Father vanished during refinement.
-    BOOST_CHECK( father.mightVanish() );
-
+    const auto& origin = element.getOrigin();
+    const auto& originLevelData = grid.currentData()[origin.level()];
     // getOrigin() returns the level-zero ancestor, even when the grid
     // has undergone multiple refinements.
-    BOOST_CHECK_EQUAL( element.getOrigin().level(), 0);
+    BOOST_CHECK_EQUAL( origin.level(), 0);
+
+    const auto& father = element.father();
+    const auto& fatherLevelData =  grid.currentData()[father.level()];
+    BOOST_CHECK( father.level() <= (isNested? grid.maxLevel() : preAdaptMaxLevel) );
+    
+    BOOST_CHECK_EQUAL( originLevelData->getMark(origin), 1);
+    BOOST_CHECK_EQUAL( fatherLevelData->getMark(father), 1);
+
+    BOOST_CHECK( !origin.isLeaf() );
+    BOOST_CHECK( origin.mightVanish() );
+    
+    BOOST_CHECK( !father.isLeaf() ); // Father vanished during refinement.
+    BOOST_CHECK( father.mightVanish() );
 
     auto itFather = father.hbegin(grid.maxLevel());
     const auto& endItFather = father.hend(grid.maxLevel());
     // If itFather != endItFather and !father.isLeaf() (if dristibuted_data_ is empty).
     BOOST_CHECK( itFather != endItFather );
-    checkReferenceElemParentCellVolume( itFather, endItFather);
-    checkReferenceElemParentCellCenter(itFather, endItFather, expected_total_children);
+    checkReferenceElemParentCellVolume(itFather, endItFather, isNested);
+    checkReferenceElemParentCellCenter(itFather, endItFather, expected_total_children,
+                                       isNested, grid.maxLevel());
+
+    auto itOrigin = origin.hbegin(grid.maxLevel());
+    const auto& endItOrigin = origin.hend(grid.maxLevel());
+    // If itOrigin != endItOrigin and !origin.isLeaf() (if dristibuted_data_ is empty).
+    BOOST_CHECK( itOrigin != endItOrigin );
```

Comment:
Concerning the comment above we could do
```suggestion
    // If itOrigin != endItOrigin and !origin.isLeaf() (if dristibuted_data_ is empty).
    BOOST_CHECK( iorigin.hbegin(grid.maxLevel() != origin.hbegin(grid.maxLevel() );
    // next level l is used to iterate only over one level
    auto l = iorigin.hbegin(grid.maxLevel())->level();
    auto itOrigin = origin.hbegin(l);
    const auto& endItOrigin = origin.hend(l);
```